test/combinatorics.jl

```diff
@@ -77,17 +77,12 @@ end
 end
 
 @testset "factorial" begin
-    @test factorial(7) == 5040
-    @test factorial(Int8(7)) == 5040
-    @test factorial(UInt8(7)) == 5040
-    @test factorial(Int16(7)) == 5040
-    @test factorial(UInt16(7)) == 5040
-    @test factorial(Int32(7)) == 5040
-    @test factorial(UInt32(7)) == 5040
-    @test factorial(Int64(7)) == 5040
-    @test factorial(UInt64(7)) == 5040
-    @test factorial(Int128(7)) == 5040
-    @test factorial(UInt128(7)) == 5040
+    for T = Base.uniontypes(Union{Base.Checked.SignedInt,Base.Checked.UnsignedInt})
+        @testset let T = T
+            @test factorial(T(7)) == 5040
+            @test Core.Compiler.is_foldable(Base.infer_effects(factorial, (T,)))
+        end
+    end
```

Comment:
```suggestion
    types = Base.uniontypes(Union{Base.Checked.SignedInt,Base.Checked.UnsignedInt})
    @testset for T = types
        @test factorial(T(7)) == 5040
        @test Core.Compiler.is_foldable(Base.infer_effects(factorial, (T,)))
    end
```

Reply:
I'd prefer the contextual testset.

Reply:
The `@testset for`  will also show the iterated object as context on a failure:

```
julia> using Test

julia> types = Base.uniontypes(Union{Base.Checked.SignedInt,Base.Checked.UnsignedInt});

julia> @testset begin
       @testset for T = types
           @test factorial(T(7)) == 5040
           @test Core.Compiler.is_foldable(Base.infer_effects(factorial, (T,)))
       end
       end

[...]
Test Summary: | Pass  Fail  Total  Time
test set      |   10    10     20  0.4s
  T = Int128  |    1     1      2  0.0s
  T = Int16   |    1     1      2  0.0s
  T = Int32   |    1     1      2  0.0s
  T = Int64   |    1     1      2  0.0s
  T = Int8    |    1     1      2  0.0s
  T = UInt128 |    1     1      2  0.0s
  T = UInt16  |    1     1      2  0.0s
  T = UInt32  |    1     1      2  0.0s
  T = UInt64  |    1     1      2  0.0s
  T = UInt8   |    1     1      2  0.0s
ERROR: Some tests did not pass: 10 passed, 10 failed, 0 errored, 0 broken.
```

(run on a ~month old master, hence the failures)